src/librustc_typeck/check/mod.rs

```diff
@@ -550,43 +548,6 @@ impl<'a, 'tcx> Visitor<'tcx> for CheckItemTypesVisitor<'a, 'tcx> {
     }
 }
 
-impl<'a, 'tcx> ItemLikeVisitor<'tcx> for CheckItemBodiesVisitor<'a, 'tcx> {
```

Comment:
You should also remove the `item_tables` queries from `CheckItemTypesVisitor` (the rest of which belongs in `collect` IMO).